File: src/java/xt/xt/audio/Callbacks.java
package xt.audio;

import xt.audio.Structs.XtBuffer;

public interface Callbacks {

    interface XtOnError  {
        void callback(String message) throws Exception;
    }

    interface XtOnXRun {
        void callback(XtStream stream, int index, Object user) throws Exception;
    }

    interface XtOnBuffer {
        int callback(XtStream stream, XtBuffer buffer, Object user) throws Exception;
    }

    interface XtOnRunning {
        void callback(XtStream stream, boolean running, long error, Object user) throws Exception;
    }
}

File: src/net/xt/Callbacks.cs
using System;
using System.Security;

namespace Xt
{
    [SuppressUnmanagedCodeSecurity]
    delegate void OnXRun(IntPtr stream, int index, IntPtr user);
    [SuppressUnmanagedCodeSecurity]
    delegate int OnBuffer(IntPtr stream, in XtBuffer buffer, IntPtr user);
    [SuppressUnmanagedCodeSecurity]
    public delegate void XtOnError(in XtLocation location, string message);
    [SuppressUnmanagedCodeSecurity]
    delegate void OnRunning(IntPtr stream, int running, ulong error, IntPtr user);

    public delegate void XtOnXRun(XtStream stream, int index, object user);
    public delegate int XtOnBuffer(XtStream stream, in XtBuffer buffer, object user);
    public delegate void XtOnRunning(XtStream stream, bool running, ulong error, object user);
}